lis/dataassim/obs/NASA_SMAPsm/read_NASASMAPsm.F90

```diff
@@ -863,22 +879,38 @@ subroutine read_SMAPL2sm_data(n, k,fname, smobs_inp, time)
         sm_data(ease_col(t) + &
              (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = sm_field(t)
 
-        if(vwc_field(t).gt.5) then !YK Aply QC : if VWC > 5 kg/m2
-           sm_data(ease_col(t) + &
-                (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = LIS_rc%udef
-        else
-           if(sm_data(ease_col(t) + &
-                   (ease_row(t)-1)*NASASMAPsm_struc(n)%nc).ne.-9999.0) then
-              if(ibits(sm_qa(t),0,1).eq.0) then
-                 sm_data_b(ease_col(t) + &
-                    (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = .true.
-              else
-                 sm_data(ease_col(t) + &
-                      (ease_row(t)-1)*NASASMAPsm_struc(n)%nc) = LIS_rc%udef
-              endif
-           endif
-        endif
-     endif
+        ! ZH: commented out VWC if-statement since it is redundant:
+        ! ibits(sm_qa(t),0,1) will equal 1 if VWC > 5 kg/m2
+        !
+        !if(vwc_field(t).gt.5) then !YK Aply QC : if VWC > 5 kg/m2
```

Comment:
so the vwc_field is not used as a condition in the new code, which is OK for the current baseline option, but if the baseline retrieval would be linked to another option (in which VWC is also retrieved), it might be good to have an extra check on the vwc_field in the case of conservative vegetation screening.

Reply:
Ok, I will include the if-statement again, conditional on `vegetation_flag.eq.1`.